src/item.cpp

```diff
@@ -312,7 +312,11 @@ item::item( const itype *type, time_point turn, int qty ) : type( type ), bday(
     }
 
     if( qty >= 0 ) {
-        charges = qty;
+        if( count_by_charges() ) {
```

Comment:
Using `type->can_have_charges()` here might be safer. But tbh I'm not sure if there's cases where this is relevant. (Yes, having charges and being count_by_charges aren't necessarily the same thing.)